src/services/publishing/publishing.ts

```diff
@@ -1,22 +1,39 @@
 import { PublishingJob } from "../../types";
-import { Wallet } from "ethers";
-import { DocumentStoreFactory } from "@govtechsg/document-store";
+import { Wallet, providers } from "ethers";
+import { DocumentStoreFactory, GsnCapableDocumentStoreFactory } from "@govtechsg/document-store";
 import { DocumentStore } from "@govtechsg/document-store/src/contracts/DocumentStore";
 import { TitleEscrowCreatorFactory, TradeTrustERC721Factory } from "@govtechsg/token-registry";
 import { TitleEscrowCreator } from "@govtechsg/token-registry/types/TitleEscrowCreator";
+import { supportsInterface } from "./utils";
+import { getGsnRelaySigner } from "../../common/config/decrypt";
 
 const assertAddressIsSmartContract = async (address: string, wallet: Wallet): Promise<void> => {
   const code = await wallet.provider.getCode(address);
   if (code === "0x") throw new Error("Address is not a smart contract");
 };
 
+const getConnectedDocumentStore = async (
+  wallet: Wallet,
+  contractAddress: string
+): Promise<DocumentStore> => {
```

Comment:
Weird that this one works, TS detects that both has the same interface?

Reply:
Hmm with our implementation ts doesn't have a way to know which address corresponds to what type of contract until it checks the interface. For all it knows, the address could be neither and it would still connect but throw an error later on